### PR TITLE
IA-1888: Org unit location and catchment edition

### PR DESCRIPTION
**Pull request by beygorghor**
Initial bug was that you can edit a marker and a shape in the same time while editing a location of a org unit.

We discovered a lots of other bugs while fixing it.

I'll describe the correct behaviour instead of describe all the errors we got.

- if you don't have a a geo location, you can add a marker or a shape
- while adding a shape, a marker or a catchment, you cannot do something else (including cancel or save)
- Org unit shape and marker should use primary color
- catchment shape should use secondary color
- make sure the marker is not disappearing

Related JIRA tickets : IA-1888

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests

## Changes

The main issue is due to the EditGroup Class object that was not refreshing his value while react did it in his state.

## How to test

Open the detail of a org unit.
Go to map tab, click on edit
Play with the location, shape and catchment tools.

## Print screen / video

https://user-images.githubusercontent.com/12494624/222453897-edf02166-3d75-44bd-affc-a1185c40924f.mov

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/orgUnits/details.js [hunk 1]
```diff
@@ -367,6 +367,7 @@ const OrgUnitDetail = ({ params, router }) => {
         isNewOrgunit,
         sourcesSelected,
     ]);
+    console.log('currentOrgUnit', currentOrgUnit);
```
Comment (codeKameleon) [inline hat/assets/js/apps/Iaso/domains/orgUnits/details.js]: Please remove the console log

**Review by codeKameleon — COMMENTED** [review]
The feature works well ! Just when you cancel an action, there is an error in the console. It happens because `pane` is undefined 

@beygorghor @quang-le Also the behavior after cancel is taking the user back to the Filters tab, i don't know if it's the behavior expected. Maybe it would be better to stay on the edit tab ? 


https://user-images.githubusercontent.com/25134301/222769026-2ce042fb-c6ea-4643-a76c-d89c812eb702.mov

**Comment by beygorghor** [comment]
> 

Yes this is the current behaviour already on prod, the tab is not deep linked

**Review by quang-le — CHANGES_REQUESTED** [review]
The "Use this location" button in the marker's pop up messes up with this feature:
- If I use it while editing a shape, the map gets stuck in a state where you can't leave the "Edit" tab
https://user-images.githubusercontent.com/38907762/223186922-660dd101-497e-419a-b376-c1b9104b5cc5.mov

- If a location shape exists and you sue the button, you can still add a marker:
https://user-images.githubusercontent.com/38907762/223186942-67f47fcf-01ab-4d59-b3a0-25dd46d24cd8.mov

I don't know if it should be blocking for the current PR, but if not we should open tickets for this

**Review by quang-le — APPROVED** [review]
The issues in my previous feedback will be handled in a separate ticket

**Pull request merged**